matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/mapper/EventAnnotationsSummaryMapper.kt

```diff
@@ -36,13 +37,22 @@ internal object EventAnnotationsSummaryMapper {
                     )
                 },
                 editSummary = annotationsSummary.editSummary
-                        ?.let {
-                            val latestEdition = it.editions.maxByOrNull { editionOfEvent -> editionOfEvent.timestamp } ?: return@let null
+                        ?.let { summary ->
+                            /**
+                             * The most recent event is determined by comparing origin_server_ts;
+                             * if two or more replacement events have identical origin_server_ts,
+                             * the event with the lexicographically largest event_id is treated as more recent.
+                             */
+                            val latestEdition = summary.editions.sortedWith(compareBy<EditionOfEvent> { it.timestamp }.thenBy { it.eventId })
```

Comment:
I am not sure the implementation is correct. I think the natural order of sorting will be the increasing order and here we would like decreasing order so that we get the most recent first. And I am also not sure about the default sort for strings. 
This is one of the reasons, I strongly think we need unit tests to validate the implementation.

What do you think?

Reply:
added tests